expo/tests/gbe/test_assign_volunteer.py

```
@@ -0,0 +1,275 @@
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.client = Client()
+        self.performer = PersonaFactory.create()
+        self.privileged_profile = ProfileFactory.create()
+        self.privileged_user = self.privileged_profile.user_object
+        group, nil = Group.objects.get_or_create(name='Volunteer Coordinator')
+        self.privileged_user.groups.add(group)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_bad_user(self):
+        ''' user does not have Volunteer Coordinator, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = ProfileFactory.create().user_object
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_no_profile(self):
+        ''' user does not have a profile, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = UserFactory.create()
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    def test_assign_volunteer_no_events(self):
+        ''' assign a volunteer when there are no opportunities'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = self.privileged_user
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+        nt.assert_equal(response.status_code, 200)
+        nt.assert_true('Assign Volunteer to Opportunities' in response.content)
+
+    def test_assign_volunteer_show_current_events(self):
```

Simplest definition: one assert per test. This is too simplistic, of course, there can be reasonable reasons to use two or even three asserts to nail down a test, as above, but it's a good target. 
More nuanced definition: tests should be sufficiently focused that a failure of one thing does not mask other failures. For example, in this set of tests, if the string "Security/usher" does not appear in the response.content, then we have no information about any of the following asserts. This is a problem, since those conditions would give valuable information in diagnosing and fixing the problems, and also because it turns test fixing into a serial process - you can only fix one thing at a time, then you have to run the tests again and fix the next thing, when you might have got them all at once if you'd had the information.